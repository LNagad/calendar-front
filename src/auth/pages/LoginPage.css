.authBody {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  
  background: #0575E6;  /* fallback for old browsers */
  background: -webkit-linear-gradient(to right, #021B79, #0575E6);  /* Chrome 10-25, Safari 5.1-6 */
  background: linear-gradient(to right, #021B79, #0575E6); /* W3C, IE 10+/ Edge, Firefox 16+, Chrome 26+, Opera 12+, Safari 7+ */
}

.authBody2 {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  
  background: #0575E6; 
  background: -webkit-linear-gradient(to right, #0575E6, #021B79); 
  background: linear-gradient(to right, #0575E6, #021B79); 
}



.login-form-1 {
  background: #fff;
  border-radius: 5px;
  box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.2), 0 9px 26px 0 rgba(0, 0, 0, 0.19);
  padding: 5%;
  min-width: 300px; /* Ajusta el min-width según tus preferencias */
}

.login-form-1 h3 {
  text-align: center;
  color: #0062cc;
  font-size: 35px; /* Tamaño de fuente ajustado a 19px */
}

.login-form-2 {
  border-radius: 5px;
  box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.2), 0 9px 26px 0 rgba(0, 0, 0, 0.19);
  padding: 5%;
  background: #fff;
  min-width: 300px; /* Ajusta el min-width según tus preferencias */
}

.login-form-2 h3 {
  text-align: center;
  color: #0062cc;
  font-size: 35px; /* Tamaño de fuente ajustado a 19px */
}

.login-container form {
  padding: 10%;
}

.btnSubmit {
  width: 100%;
  border-radius: 1rem;
  padding: 1.5%;
  border: none;
  cursor: pointer;
  font-size: 19px; /* Tamaño de fuente ajustado a 19px */
}

.login-form-1 .btnSubmit {
  color: #fff;
  background-color: #0062cc;
}

.login-form-2 .btnSubmit {
  background-color: #fff;
  color: #0062cc;
}

.login-form-2 .ForgetPwd {
  color: #fff;
  font-weight: 600;
  text-decoration: none;
}

.login-form-1 .ForgetPwd {
  color: #0062cc;
  font-weight: 600;
  text-decoration: none;
}

.form-group input.form-control {
  font-size: 19px; /* Tamaño de fuente ajustado a 19px */
}

.form-group .invalid-feedback {
  font-size: 19px; /* Tamaño de fuente ajustado a 19px */
}

.mt-2.text-primary {
  font-size: 19px; /* Tamaño de fuente ajustado a 19px */
}

.text-primary.ms-1 {
  font-size: 19px; /* Tamaño de fuente ajustado a 19px */
}

@media (max-width: 767px) {
  .authBody2 {
    justify-content: start;
  }
  .authBody2 .login-container{
    margin-top: 6rem;
  }

  .login-container {
    height: auto; /* Altura automática en dispositivos móviles */
    padding: 20px; /* Espaciado adicional en dispositivos móviles */
  }

  .form-group {
    margin-bottom: 1rem !important;
  }

} 

