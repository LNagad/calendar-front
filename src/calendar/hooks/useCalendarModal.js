import { addHours, differenceInSeconds } from 'date-fns';
import { useMemo, useState } from 'react';
import Swal from 'sweetalert2';
import { useCalendarStore, useUIStore } from '../../hooks';
import { useEffect } from 'react';

export const useCalendarModal = () => {

   const { activeEvent, startSavingEvent } = useCalendarStore();
   const { isDateModalOpen, closeDateModal } = useUIStore();
   const [formSubmitted, setFormSubmitted] = useState(false);
    
   const [formValues, setFormValues] = useState({
      title: '',
      notes: '',
      start: new Date(),
      end: addHours( new Date(), 2),
   });

   useEffect(() => {
      if ( activeEvent !== null ) {
         setFormValues({ ...activeEvent });
      }
      
   }, [ activeEvent ]);
   
    
   const titleClass = useMemo(() => {
      if ( !formSubmitted ) return '';
    
      return ( formValues.title.length > 0 )
         ? 'is-valid'
         : 'is-invalid';
       
   }, [ formValues.title, formSubmitted ]);
    
   const onInputChange = ({target}) => {
      setFormValues({
         ...formValues,
         [target.name]: target.value
      });
   };
    
   const onDateChange = ( event, changing ) => {
      setFormValues({
         ...formValues,
         [changing]: event
      });
   };
    
   const onCloseModal = () => {
      closeDateModal();
      // setIsModalOpen(false);
   };
    
   const onSubmit = async( event ) => {
      event.preventDefault();
      setFormSubmitted(true);
    
      const difference = differenceInSeconds( formValues.end, formValues.start );
    
      if ( isNaN( difference ) || difference <= 0 ) {
         Swal.fire('Error en fechas', 'Fechas incorrectas', 'error' );
         return;
      }
    
      if ( formValues.title.length <= 0 ) return;
    
      await startSavingEvent( formValues );
      closeDateModal();
      setFormSubmitted(false);
   };

   return {
      isModalOpen: isDateModalOpen,
      onSubmit,
      onCloseModal,
      onDateChange,
      onInputChange,
      titleClass,
      
      formValues
   };
};
