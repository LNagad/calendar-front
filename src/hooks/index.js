export * from './useUIStore';
export * from './useAuthStore';
export * from './useForm';
export * from './useCalendarStore';