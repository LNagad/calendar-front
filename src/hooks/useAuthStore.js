import { useDispatch, useSelector } from 'react-redux';
import { calendarApi } from '../api';
import { clearErrorMessage, onChecking, onLogin, onLogout, onLogoutCalendar, toggleBtnLoginSpinner, toggleBtnRegisterSpinner } from '../store';

export const useAuthStore = () => {
   
   const { status, user, errorMessage } = useSelector( state => state.auth);
   const {
      btnLoginSpinnerIsLoading, 
      btnRegisterSpinnerIsLoading 
   } = useSelector( state => state.ui);

   const dispatch = useDispatch();
   
   const startLogin = async({ email, password }) => {
      dispatch( onChecking() );
      dispatch( toggleBtnLoginSpinner() );

      try {
        
         const { data } = await calendarApi.post('/auth/login', { email, password});
         localStorage.setItem('token', data.token);
         localStorage.setItem('token-init-date', new Date().getTime() );
         dispatch( onLogin({ name: data.name, _id: data.uid }) );
         dispatch( toggleBtnLoginSpinner() );

      } catch (error) {
         console.log(error);
         localStorage.clear();
         dispatch( onLogout(error?.response?.data?.msg || 'Invalid credentials') );
         dispatch( toggleBtnLoginSpinner() );

         setTimeout(() => {
            dispatch( clearErrorMessage() );
         }, 10);
      }

   };

   const startRegistering = async({ registerName, registerEmail , registerPassword }) => {
      dispatch( onChecking() );
      dispatch( toggleBtnRegisterSpinner() );
     
      try {
         const { data } = await calendarApi.post('/auth/signup', {
            name: registerName,
            email: registerEmail.toLowerCase(),
            password: registerPassword
         });

       
         localStorage.setItem('token', data.token);
         localStorage.setItem('token-init-date', new Date().getTime() );
         dispatch( onLogin({ name: data.name, _id: data.uid }) );
         dispatch( toggleBtnRegisterSpinner() );
     
   
      } catch (error) {
         dispatch( onLogout(error?.response?.data?.msg || '--') );
         dispatch( toggleBtnRegisterSpinner() );

         setTimeout(() => {
            dispatch( clearErrorMessage() );
         }, 10);
      }

   };

   const startLogout = async() => {
      localStorage.clear();
      dispatch( onLogout() );
      dispatch( onLogoutCalendar() );
   };

   const checkAuthToken = async () => {
      const token = localStorage.getItem('token');
      if ( !token ) return dispatch( onLogout() );

      try {
         
         const { data } = await calendarApi.get('auth/renew');
         localStorage.setItem('token', data.token);
         localStorage.setItem('token-init-date', new Date().getTime() );
         dispatch( onLogin({ name: data.name, _id: data.uid }) );
     
      } catch (error) {
         localStorage.clear();
         dispatch( onLogout() );
      }
   };

   return {
      //* Properties
      status, 
      user, 
      errorMessage,
      btnLoginSpinnerIsLoading, 
      btnRegisterSpinnerIsLoading,

      //* Methods
      startLogin,
      startRegistering,
      checkAuthToken,
      startLogout

   };
};
