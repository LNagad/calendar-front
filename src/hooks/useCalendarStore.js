import { useDispatch, useSelector } from 'react-redux';
import { onAddNewEvent, onDeleteEvent, onLoadEvents, onSetActiveEvent, onUpdateEvent, unSelectActiveEvent } from '../store';
import { calendarApi } from '../api';
import { convertDateEvents } from '../helpers';
import Swal from 'sweetalert2';

export const useCalendarStore = () => {
   const dispatch = useDispatch();
   const { user } = useSelector( state => state.auth );
   const { events, activeEvent, isLoadingEvents } = useSelector( state => state.calendar );
   
   const setActiveEvent = ( calendarEvent ) => {
      dispatch( onSetActiveEvent( calendarEvent ) );
   };

   const onUnSelectActiveEvent = () => {
      dispatch( unSelectActiveEvent() );
   };

   const startLoadingEvents = async() => {
      try {
      
         const { data } = await calendarApi.get('/events');
         const events = convertDateEvents( data.events );
         dispatch( onLoadEvents(events) );
         
      } catch (error) {
         console.log('Error fetching events');
         console.log(error);
      }  
   };

   const startSavingEvent = async( calendarEvent ) => {

      try {
         if (calendarEvent.id){
            //* Updating
            await calendarApi.put(`/events/${ calendarEvent.id }`, calendarEvent);
            dispatch( onUpdateEvent({ ...calendarEvent, user }));
            return;
         } 
   
         //* Creating
         const { data } = await calendarApi.post('/events', calendarEvent);
         dispatch( onAddNewEvent({ ...calendarEvent, id: data.event.id, user }) );
            
      } catch (error) {
         console.log(error);
         Swal.fire('Couldn\'t save', error.response.data.msg, 'error');
      }
   };

   const startDeletingEvent = async() => {
      
      try {
         await calendarApi.delete(`/events/${ activeEvent.id }`);
         Swal.fire('Deleted', `${activeEvent.title} has been sucesfully removed`, 'success');
         dispatch( onDeleteEvent() );
      } catch (error) {
         console.log(error);
         Swal.fire('Couldn\'t delete this note', error.response.data.msg, 'error');
      }
   };

   return {
      //* Propiedades
      isLoadingEvents,
      events,
      activeEvent,
      hasEventSelected: !!activeEvent, //? null = false - object = true

      //* Metodos
      setActiveEvent,
      onUnSelectActiveEvent,
      startSavingEvent,
      startDeletingEvent,
      startLoadingEvents,
   };
};
