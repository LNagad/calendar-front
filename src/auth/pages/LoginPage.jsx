import { useEffect } from 'react';
import Swal from 'sweetalert2';
import { useAuthStore, useForm } from '../../hooks';
import { LoadingSpinner } from '../../ui';
import './LoginPage.css';
import { Link } from 'react-router-dom';
import { useState } from 'react';

const loginFormFields = {
   loginEmail: '',
   loginPassword: '',
};

const loginFormValidations = {
   loginPassword: [ (value) => value.length >= 5, 'The name must be at least 5 characters long'],
   loginEmail: [
      (value) => {
         const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
         if (!emailRegex.test(value)) {
            return false;
         }
         return true; // Retorna `true` si pasa todas las validaciones
      },
      'The email is invalid',
   ],
};

export const LoginPage = () => {
   const [loginSubmitted, setLoginSubmitted] = useState(false);

   const { 
      startLogin, 
      errorMessage, 
      btnLoginSpinnerIsLoading, 
   } = useAuthStore();

   const { 
      loginEmail, 
      loginPassword, 
      onInputChange: onLoginInputChange,
      isFormValid,
      loginEmailValid, 
      loginPasswordValid, 
   } = useForm( loginFormFields, loginFormValidations );
   
   const loginSubmit = ( event ) => {
      event.preventDefault();
      setLoginSubmitted(true);
      if ( !isFormValid ) return;

      startLogin({ email: loginEmail, password: loginPassword });
      setLoginSubmitted(false);
   };
   
   useEffect(() => {
      if ( errorMessage !== undefined ) {
         Swal.fire('Authentication error', errorMessage, 'error');
      }
   }, [errorMessage]);

   
   return (
      <section className='authBody animate__animated animate__fadeIn animate__fast'>
         <div className="container login-container animate__animated animate__fadeIn animate__slower">
            <div className="row d-flex justify-content-center">
               <div className="col-md-6 login-form-1">
                  <h3>Ingreso</h3>
                  <form onSubmit={ loginSubmit }>
                     {
                        (errorMessage) &&
                     <div className="form-group mb-2">
                        <div className="alert alert-danger alert-dismissible fade show" role="alert">
                           <strong>{errorMessage} </strong> 
                           You should check in on some of those fields below.
                           <button type="button" className="btn-close" data-bs-dismiss="alert" aria-label="Close"></button>
                        </div>
                     </div>
                     }
                     <div className="form-group mb-2">
                        <input
                           autoComplete='off' 
                           type="email"
                           className={`form-control ${ (loginSubmitted) && !!loginEmailValid && 'is-invalid'}`}
                           placeholder="Correo"
                           name='loginEmail'
                           value={ loginEmail }
                           onChange={ onLoginInputChange }
                        />
                        <div className="invalid-feedback mb-2">
                           {loginEmailValid}
                        </div>
                     </div>
                     <div className="form-group mb-2">
                        <input
                           autoComplete='off'
                           type="password"
                           className={`form-control ${ (loginSubmitted) && !!loginPasswordValid && 'is-invalid'}`}
                           placeholder="ContraseÃ±a"
                           name='loginPassword'
                           value={ loginPassword }
                           onChange={ onLoginInputChange }
                        />
                        <div className="invalid-feedback mb-2">
                           {loginPasswordValid}
                        </div>
                     </div>
                     <div className="form-group mb-2 position-relative">
                        <button 
                           disabled={btnLoginSpinnerIsLoading}
                           type="submit"
                           className="btn btn-primary rounded-4"
                           value="Login" 
                        >
                           <span className='fw-semibold fs-5'>Login</span>
                        </button>
                        {
                           (btnLoginSpinnerIsLoading) &&
                         <LoadingSpinner color='text-light' />
                        }
                     </div>
                     <div className="mt-2 text-primary float-end">
                        <span>Don`t have an account?</span>
                        <Link to={'/auth/register'} 
                           className='text-primary ms-1'>create one</Link>
                     </div>
                  </form>
               </div>
      
           
            </div>
         </div>
      </section>
   );
};