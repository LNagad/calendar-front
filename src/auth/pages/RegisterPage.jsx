import { useEffect, useState } from 'react';
import Swal from 'sweetalert2';
import { useAuthStore, useForm } from '../../hooks';
import { LoadingSpinner } from '../../ui';
import './LoginPage.css';
import { Link } from 'react-router-dom';


const registerFormFields = {
   registerName: '',
   registerEmail: '',
   registerPassword: '',
   registerPassword2: '',
};

const registerFormValidations = {
   registerName: [ (value) => value.length >= 5, 'The name must be at least 5 characters long'],
   registerEmail: [
      (value) => {
         // if (!value.includes('@') || !value.includes('.com')) {
         //    return false;
         // }
         // if (value.includes(' ')) {
         //    return false;
         // }
         const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
         if (!emailRegex.test(value)) {
            return false;
         }
         return true; // Retorna `true` si pasa todas las validaciones
      },
      'The emailis invalid',
   ],
   registerPassword: [ (value) => value.length >= 10, 'The name must be at least 10 characters long'],
   // registerPassword2: '',
};

export const RegisterPage = () => {
   const { 
      startRegistering,
      errorMessage,  
      btnRegisterSpinnerIsLoading, 
   } = useAuthStore();

   const [registerSubmitted, setRegisterSubmitted] = useState(false);

   const { 
      registerName, 
      registerEmail, 
      registerPassword, 
      registerPassword2, 
      onInputChange: onRegisterInputChange,
      //validations
      isFormValid,
      registerNameValid, 
      registerEmailValid, 
      registerPasswordValid, 
   } = useForm( registerFormFields, registerFormValidations );

   const registerPasswordValid2 = registerPassword === registerPassword2 && registerPassword2.length >= 10;
   
   const registerSubmit = ( event ) => {
      event.preventDefault();
      setRegisterSubmitted(true);
      
      if ( !isFormValid ) {
         Swal.fire('Invalid fields', 'Please complete all the fields', 'error');
         return;
      }
      
      if ( !registerPasswordValid2 ) {
         Swal.fire('Invalid password', 'The password doesn\'t match', 'error');
         return;
      }

      
      startRegistering({ registerName, registerEmail, registerPassword });
   };

   useEffect(() => {
      if ( errorMessage !== undefined ) {
         Swal.fire('Authentication error', errorMessage, 'error');
      }
   }, [errorMessage]);

   
   return (
      <section className='authBody2 animate__animated animate__fadeIn animate__fast'>
         <div className="container login-container animate__animated animate__fadeIn animate__slower">
            <div className="row d-flex justify-content-center">
               <div className="col-md-6 login-form-2">
                  <h3>Registro</h3>
                  <form onSubmit={ registerSubmit }>
                     <div className="form-group mb-2">
                        <input
                           autoComplete='off'
                           type="text"
                           className={`form-control ${ (registerSubmitted) && registerNameValid && 'is-invalid'}`}
                           placeholder="Nombre"
                           name='registerName'
                           value={ registerName }
                           onChange={ onRegisterInputChange }
                        />
                     </div>
                     <div className="form-group mb-2">
                        <input
                           autoComplete='off'
                           type="email"
                           className={`form-control ${ (registerSubmitted) &&  registerEmailValid && 'is-invalid'}`}
                           placeholder="Correo"
                           name='registerEmail'
                           value={ registerEmail }
                           onChange={ onRegisterInputChange }
                        />
                     </div>
                     <div className="form-group mb-2">
                        <input
                           autoComplete='off'
                           type="password"
                           className={`form-control ${ (registerSubmitted) && registerPasswordValid && 'is-invalid'}`}
                           placeholder="Contraseña" 
                           name='registerPassword'
                           value={ registerPassword }
                           onChange={ onRegisterInputChange }
                        />
                     </div>
      
                     <div className="form-group mb-2">
                        <input
                           autoComplete='off'
                           type="password"
                           className={`form-control ${(registerSubmitted) && !registerPasswordValid2 && 'is-invalid'}`}
                           placeholder="Repita la contraseña" 
                           name='registerPassword2'
                           value={ registerPassword2 }
                           onChange={ onRegisterInputChange }
                        />
                     </div>
      
                     <div className="form-group mb-2 position-relative">
                        <button 
                           disabled={ (registerSubmitted) && !isFormValid || btnRegisterSpinnerIsLoading}
                           type="submit"
                           className="btn btn-primary rounded-4"
                           value="Login" 
                        >
                           <span className='fw-semibold text-light fs-5'>Register</span>
                        </button>
                        {
                           (btnRegisterSpinnerIsLoading) &&
                          <LoadingSpinner color='text-primary' />
                        }
                     </div>
                     <div className="mt-2 text-primary float-end">
                        <span>Already have an account? Return to</span>
                        <Link to={'/auth/login'} 
                           className='text-primary ms-1'>login</Link>
                     </div>
                  </form>
                 
               </div>
            </div>
         </div>
      </section>
   );
};