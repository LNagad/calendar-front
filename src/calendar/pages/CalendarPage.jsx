import { useState } from 'react';
import { Calendar } from 'react-big-calendar';
import 'react-big-calendar/lib/css/react-big-calendar.css';

import { CalendarEvent, CalendarModal, FabAddNew, FabDelete, Navbar } from '../';
import { getMessagesES, localizer } from '../../helpers';
import { useAuthStore, useCalendarStore, useUIStore } from '../../hooks';
import { useEffect } from 'react';
import { CalendarSpinner } from '../../ui';

// import withDragAndDrop from 'react-big-calendar/lib/addons/dragAndDrop';
// const DnDCalendar = withDragAndDrop(Calendar);

export const CalendarPage = () => {
   
   const { user } = useAuthStore();
   const { openDateModal } = useUIStore();
   const { isLoadingEvents, activeEvent, events, setActiveEvent, onUnSelectActiveEvent, startLoadingEvents } = useCalendarStore();
   
   const [lastView, setLastView] = useState(localStorage.getItem('lastView') || 'month');

   const eventStyleGetter = ( event, start, end, isSelected ) => {
      const isMyEvent = ( user._id === event.user._id );

      const style = {
         backgroundColor: isMyEvent ? '#347CF7' : '#465660',
         borderRadius: '0px',
         opacity: 1,
         color: 'white'
      };
   
      return {
         style
      };
   };
   
   /**
    * TODO: usar onSelectSlot para agregar un evento de forma 
    * TODO: mas reactiva al clickear un slot del calendar
    */

   const onSelectSlot = ( { action, start } ) => {
      
      if ( action === 'click' ) {
         onUnSelectActiveEvent();
      }
      // console.log({ action, start});
   };
   
   const onDoubleClick = ( event ) => {
      // console.log({ doubleClick: event});
      if ( event.user._id !== user._id ) {
         return;
      }
      openDateModal();
   };
   
   const onSelect = ( event ) => {
      // console.log({ click: event});
      if ( event.user._id !== user._id ) {
         if (activeEvent) {
            onUnSelectActiveEvent(); 
         }
         return;
      }
      setActiveEvent( event );
   };
   
   const onViewChange = ( event ) => {
      // console.log({ viewChanged: event});
      localStorage.setItem('lastView', event);
      setLastView(event);
   };

   useEffect(() => {
      startLoadingEvents();
   }, []);
   

   return (
      <>
         <Navbar />
         {
            ( isLoadingEvents ) 
               ? <CalendarSpinner color='text-primary' />
               :
               <>
                  <Calendar
                     culture='es'
                     localizer={localizer}
                     events={events}
                     defaultView={ lastView }
                     startAccessor="start"
                     endAccessor="end"
                     style={{ height: 'calc( 100vh - 105px ) ' }}
                     messages={ getMessagesES() }
                     eventPropGetter={ eventStyleGetter }
                     components={{ event: CalendarEvent }}
                     onDoubleClickEvent={ onDoubleClick }
                     onSelectEvent={ onSelect }
                     onView={ onViewChange }
                     onSelectSlot={ onSelectSlot }
                     selectable
                  />

                  <CalendarModal />
                  <FabAddNew />
                  <FabDelete />
               </>             
         }
      </>
   );
};
