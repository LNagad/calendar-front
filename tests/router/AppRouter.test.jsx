import { render, screen } from '@testing-library/react';
import { useAuthStore } from '../../src/hooks/useAuthStore';
import { AppRouter } from '../../src/router/AppRouter';
import { MemoryRouter } from 'react-router-dom';
import { CalendarPage } from '../../src/calendar/pages/CalendarPage';

jest.mock('../../src/hooks/useAuthStore');
jest.mock('../../src/calendar/pages/CalendarPage', () => ({
   CalendarPage: () => <h1>Calendar Page</h1>
}));

describe('testing AppRouter', () => { 
   
   const mockCheckAuthToken = jest.fn();

   beforeEach(() => jest.clearAllMocks() );

   test('should show the login if not authenticated', () => { 
      //   jest.fn().mockReturnValue;
    
      useAuthStore.mockReturnValue({
         status: 'not-authenticated', checkAuthToken: mockCheckAuthToken
      });

      const { container } =render( <AppRouter />,{
         wrapper: ({children}) => <MemoryRouter>{children}</MemoryRouter>
      });

      expect( screen.getByText('Ingreso') ).toBeTruthy();
      expect( container ).toMatchSnapshot();
   });

   test('should show the calendar if user is authenticated', () => { 
      useAuthStore.mockReturnValue({
         status: 'authenticated', checkAuthToken: mockCheckAuthToken
      });

      render( <AppRouter />,{
         wrapper: ({children}) => <MemoryRouter>{children}</MemoryRouter>
      });

      expect( screen.getByText('Calendar Page') ).toBeTruthy();
   });
});