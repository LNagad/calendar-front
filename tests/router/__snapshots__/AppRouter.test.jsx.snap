// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing AppRouter should show the login if not authenticated 1`] = `
<div>
  <section
    class="authBody animate__animated animate__fadeIn animate__fast"
  >
    <div
      class="container login-container animate__animated animate__fadeIn animate__slower"
    >
      <div
        class="row d-flex justify-content-center"
      >
        <div
          class="col-md-6 login-form-1"
        >
          <h3>
            Ingreso
          </h3>
          <form>
            <div
              class="form-group mb-2"
            >
              <input
                autocomplete="off"
                class="form-control false"
                name="loginEmail"
                placeholder="Correo"
                type="email"
                value=""
              />
              <div
                class="invalid-feedback mb-2"
              >
                The email is invalid
              </div>
            </div>
            <div
              class="form-group mb-2"
            >
              <input
                autocomplete="off"
                class="form-control false"
                name="loginPassword"
                placeholder="ContraseÃ±a"
                type="password"
                value=""
              />
              <div
                class="invalid-feedback mb-2"
              >
                The name must be at least 5 characters long
              </div>
            </div>
            <div
              class="form-group mb-2 position-relative"
            >
              <button
                class="btn btn-primary rounded-4"
                type="submit"
                value="Login"
              >
                <span
                  class="fw-semibold fs-5"
                >
                  Login
                </span>
              </button>
            </div>
            <div
              class="mt-2 text-primary float-end"
            >
              <span>
                Don\`t have an account?
              </span>
              <a
                class="text-primary ms-1"
                href="/auth/register"
              >
                create one
              </a>
            </div>
          </form>
        </div>
      </div>
    </div>
  </section>
</div>
`;
